pennylane/math/quantum.py

```diff
@@ -281,15 +290,20 @@ def _batched_partial_trace(density_matrix, indices):
     """
     # Autograd does not support same indices sum in backprop, and tensorflow
     # has a limit of 8 dimensions if same indices are used
-    if get_interface(density_matrix) in ["autograd", "tensorflow"]:
-        return _batched_partial_trace_nonrep_indices(density_matrix, indices)
+    matrix = cast(matrix, dtype=c_dtype)
+    if qml.math.ndim(matrix) == 2:
+        batch_dim, dim = 1, matrix.shape[1]
+    else:
+        batch_dim, dim = matrix.shape[:2]

```

Comment:
```suggestion
    if qml.math.ndim(matrix) == 2:
        is_batched = False
        batch_dim, dim = 1, matrix.shape[1]
    else:
        is_batched = True
        batch_dim, dim = matrix.shape[:2]

```

the `is_batched` flag indicates whether or not to return a matrix or a list of matrices